workflow/executor/executor.go

```diff
@@ -425,14 +425,24 @@ func (we *WorkflowExecutor) SaveParameters() error {
 			log.Infof("Copying %s from base image layer", param.ValueFrom.Path)
 			output, err = we.RuntimeExecutor.GetFileContents(mainCtrID, param.ValueFrom.Path)
 			if err != nil {
-				return err
+				// We have a default value to use instead of returning an error
+				if param.ValueFrom.Default != "" {
+					output = param.ValueFrom.Default
+				} else {
+					return err
+				}
 			}
 		} else {
 			log.Infof("Copying %s from from volume mount", param.ValueFrom.Path)
 			mountedPath := filepath.Join(common.ExecutorMainFilesystemDir, param.ValueFrom.Path)
 			out, err := ioutil.ReadFile(mountedPath)
 			if err != nil {
-				return err
+				// We have a default value to use instead of returning an error
+				if param.ValueFrom.Default != "" {
```

Comment:
test?

Reply:
I didn't include tests for this because it would be difficult to mock this behavior and I have already included a non-E2E test that covers the same behavior here: https://github.com/argoproj/argo/pull/2500/files#diff-4ed679a69dc8fdd934fb1e7f7c0f3259R109-R139

If you think it's necessary, I can invest the time into mocking this or creating an e2e test